packages/run-protocol/src/vpool-xyk-amm/multipoolMarketMaker.js

```diff
@@ -193,6 +193,7 @@ const start = async (zcf, privateArgs) => {
     }
 
     const pool = isSecondary(brandOut) ? getPool(brandOut) : getPool(brandIn);
+    // @ts-expect-error cast
```

Comment:
@Chris-Hibbert is it safe to cast to `DoublePoolInternalFacet` here?

Reply:
> renaming of `VPoolInternalFacet` to `DoublePoolInternalFacet` looks like a mistake

All properties of `VPoolInternalFacet` specified double,
```
/**
 * @typedef {Object} VPoolInternalFacet - virtual pool for price quotes and trading
 * @property {GetDoublePoolSwapQuote} getPriceForInput
 * @property {GetDoublePoolSwapQuote} getPriceForOutput
 * @property {(ZCFSeat, DoublePoolSwapResult) => string} allocateGainsAndLosses
 */
```

Is there a type for `VPoolInternalFacet` that doesn't presume pool type?

Reply:
As far as I can tell, no. So the types are not generic here, even though the code kinda is.

Reply:
It looks like someone made the type a union type, so it's now actually accurate.